programs/bubblegum/program/src/processor/finalize_tree_with_root.rs

```diff
@@ -169,7 +177,9 @@ pub(crate) fn check_stake<'info>(
         BubblegumError::StakingVoterDiscriminatorMismatch
     );
 
-    let voter: Voter = *bytemuck::from_bytes(&(*voter_bytes.borrow())[8..]);
+    // The Voter has a rather big structure. If we try to deserialize it on the stack it will be overflowed. Therefore, Box is used to move bytes that have been borrowed to the Heap.
+    let voter_bytes = Box::new(voter_bytes.borrow());
```

Comment:
I've run tests on my machine without that Box, seems like everything works. I don't think we need that Box because we don't have dereferencing anymore, therefore we do not copy anything. It means, the stack size should be increased only by several reference sizes and we should be fine with it.

Reply:
you're right, thanks for checking this!